salt/apps/ocw/sync_repo.sls

```diff
@@ -0,0 +1,31 @@
+configure_ocw_src_git_sparsecheckout:
+  git.config_set:
+    - name: core.sparsecheckout
+    - value: True
+    - cwd: /usr/local/Plone/zeocluster/src
+
+configure_ocw_publishing_git_sparsecheckout:
+  git.config_set:
+    - name: core.sparseCheckout
+    - value: True
+    - cwd: /usr/local/Plone/zeocluster/publishing
+
+add_ocw_publishing_to_sparsecheckout:
+  file.managed:
+    - name: /usr/local/Plone/zeocluster/src/.git/info/sparse-checkout
+    - contents: 'plone/publishing'
+
+add_ocw_src_to_sparsecheckout:
+  file.managed:
+    - name: /mnt/ocwfileshare/OCWEngines/.git/info/sparse-checkout
+    - contents: 'plone/src'
+
+git_pull_ocw_src_folder:
+  git.latest:
+    - name: https://github.com/mitocw/ocwcms
```

Comment:
We will need to either use an API token in the HTTPS URL for basic auth, or we will need to use SSH checkout and provision a deploy key for being able to clone from the CMS repo since it's private. (same for the next block)